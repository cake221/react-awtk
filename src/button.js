function applicationInit() {
  var win = TWindow.create(null, 0, 0, 0, 0);
  var ok = TButton.create(win, 0, 0, 0, 0);

  ok.setText("ok");
  ok.setSelfLayoutParams("center", "middle", "50%", "30");

  ok.on(TEventType.CLICK, function(evt) {
    var e = TPointerEvent.cast(evt);
    console.log("global: " + TGlobal.getPointerX() + " " + TGlobal.getPointerY() + " " + TGlobal.isPointerPressed());
    console.log("on click: " + e.x + " " + e.y);

    return TRet.OK;
  });
  
  console.log("global: " + TGlobal.getPointerX() + " " + TGlobal.getPointerY() + " " + TGlobal.isPointerPressed());

  win.layout();
}

export {
  applicationInit
}

