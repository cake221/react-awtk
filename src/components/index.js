import Button from "./Button"
import Window from "./Window"
import Node from "./Node"
import Root from "./Root"
import Global from "./Global"
import Text from "./Text"

export {
    Button,
    Window,
    Node,
    Root,
    Global,
    Text
}
