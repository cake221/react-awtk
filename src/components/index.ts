export * from "./Button"
export * from "./Window"
export * from "./CheckButton"
export * from "./Edit"
export * from "./ComboBox"
export * from "./Image"
export * from "./Label"
export * from "./RichText"
export * from "./ProgressBar"
export * from "./Slider"
export * from "./SpinBox"
