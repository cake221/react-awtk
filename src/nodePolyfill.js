var awtk = require("./native/awtk.node");
awtk.init(320, 480, 'Hello-AWTK');
awtkStepId = setInterval(awtkStep, 16);
function awtkStep() {
  if(!awtk.step()) {
    clearInterval(awtkStepId);
    console.log("awtk quit");
    awtkStepId = 0;
  }
}
