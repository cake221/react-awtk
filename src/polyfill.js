/* global globalThis */
// import * as os from 'os'
globalThis.process = { env: { NODE_ENV: 'development' } }
// globalThis.setTimeout = os.setTimeout
globalThis.setTimeout = function(callback, after, arg1, arg2, arg3){ callback() }
globalThis.console.warn = console.log
globalThis.console.error = () => {}
