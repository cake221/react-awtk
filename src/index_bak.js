import React, { Component } from 'react';
import { render } from 'react-dom';
import {applicationInit} from "./button"

class Button extends Component {
  render() {
    return <h1>Hello,wenyufei</h1>
  }
}

render(<Button/>, window.document.getElementById('app'));
