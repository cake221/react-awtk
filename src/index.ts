// import './polyfill'
import "./nodePolyfill"
export { AwtkRender } from "./renderer";
export { setParentWidget } from "./utils/fixParentChildComponent";
// todo 得区分什么可以传出去，什么不可以
// export { TRet, TWindowManager, TEventType, TGlobal, TPointerEvent, TButton, TWindow } from "./native/awtk"
export * from "./native/awtk"
export { t_button_base, t_window_base, t_check_button_base } from "./components"

