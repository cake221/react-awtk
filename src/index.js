"use strict";
exports.__esModule = true;
require("./polyfill");
var renderer_1 = require("./renderer");
exports.AwtkRender = renderer_1["default"];
var fixParentChildComponent_1 = require("./utils/fixParentChildComponent");
exports.setParentWidget = fixParentChildComponent_1.setParentWidget;
var awtk_1 = require("./native/awtk");
exports.TEventType = awtk_1.TEventType;
var t_button = "t_button";
exports.t_button = t_button;
var t_window = "t_window";
exports.t_window = t_window;
