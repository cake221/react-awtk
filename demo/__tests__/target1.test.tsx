import { Component } from 'react';
import * as React from "react"
import { AwtkRender, setParentWidget } from "../../src"

class App extends Component {
  constructor(props){
    super(props);
  }
  
  componentDidMount () {
    console.log('APP DID MOUNT!');
  }
  
  render() {
    return (
      <t_window
        ref = {
          (ref) => setParentWidget(ref, "win1")
        }
      >
        <t_button
          parent = { "win1" }
          text = { "按钮1" }
          style = {{
            selfLayout:{
              x:"center",
              y:"middle",
              w:"50%",
              h:"30"
            },
          }}
          onClick = {
            (e)=>{
              console.log("按钮1",e);
              return TRet.OK;
            }
          }
        />
        <t_button
          parent = { "win1" }
          name="open:dialog1"
          style = {{
            selfLayout:{
              x:"10",
              y:"10",
              w:"50%",
              h:"30"
            },
          }}
          text="按钮2"
          onClick = {
            (e)=>{
              console.log("按钮2",e);
              return TRet.OK;
            }
          }
        />
      </t_window>
    )
  }
}

describe("test target1", ()=>{

  it("toMatchSnapshot", ()=>{
    AwtkRender(<App />, () => {
      console.log(' 渲染成功 AWTK ');
    });
    expect(__AwtkSnapshot).toMatchSnapshot()
  })
})

