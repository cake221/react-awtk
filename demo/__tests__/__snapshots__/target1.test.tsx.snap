// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test target1 toMatchSnapshot 1`] = `
Array [
  "getRootHostContext",
  "getChildHostContext",
  "shouldSetTextContent",
  "getChildHostContext",
  "shouldSetTextContent",
  "createInstance",
  "funtion  button_create( parent, x, y, w, h ) :null, 0, 0, 0, 0",
  "funtion  widget_set_self_layout_params( widget, x, y, w, h ) :undefined, center, middle, 50%, 30",
  "funtion  widget_set_text_utf8( widget, text ) :undefined, 按钮1",
  "funtion  widget_on( widget, type, on_event, ctx ) :undefined, EVT_CLICK, function (e) {
                    console.log(\\"按钮1\\", e);
                    return TRet.OK;
                }, null",
  "finalizeInitialChildren",
  "getChildHostContext",
  "shouldSetTextContent",
  "createInstance",
  "funtion  button_create( parent, x, y, w, h ) :null, 0, 0, 0, 0",
  "funtion  widget_set_self_layout_params( widget, x, y, w, h ) :undefined, 10, 10, 50%, 30",
  "funtion  widget_set_text_utf8( widget, text ) :undefined, 按钮2",
  "funtion  widget_set_name( widget, name ) :undefined, open:dialog1",
  "funtion  widget_on( widget, type, on_event, ctx ) :undefined, EVT_CLICK, function (e) {
                    console.log(\\"按钮2\\", e);
                    return TRet.OK;
                }, null",
  "finalizeInitialChildren",
  "createInstance",
  "funtion  window_create( parent, x, y, w, h ) :null, 0, 0, 0, 0",
  "appendInitialChild",
  "appendInitialChild",
  "finalizeInitialChildren",
  "prepareForCommit",
  "appendNodeChild",
  "resetAfterCommit",
  "getPublicInstance",
  "funtion  widget_add_child( widget, child ) :undefined, [object Object]",
  "funtion  widget_add_child( widget, child ) :undefined, [object Object]",
  "funtion  widget_layout( widget ) :undefined",
]
`;
